zerver/lib/zcommand.py

```diff
@@ -5,36 +5,41 @@
 from zerver.lib.actions import do_set_user_display_setting
 from zerver.lib.exceptions import JsonableError
 
+def do_switch_setting(command: str, switch_command: str,
+                      setting: str, setting_value: bool,
+                      user_profile: UserProfile) -> str:
+    msg = 'Changed to {command} mode! To revert ' \
+          '{command} mode, type `/{switch_command}`.'.format(
+              command=command,
+              switch_command=switch_command
+          )
+    do_set_user_display_setting(user_profile=user_profile,
+                                setting_name=setting,
+                                setting_value=setting_value)
+    return msg
+
 def process_zcommands(content: str, user_profile: UserProfile) -> Dict[str, Any]:
     if not content.startswith('/'):
         raise JsonableError(_('There should be a leading slash in the zcommand.'))
+
     command = content[1:]
 
     if command == 'ping':
-        ret = dict()  # type: Dict[str, Any]
-        return ret
-
-    night_commands = ['night', 'dark']
-    day_commands = ['day', 'light']
-
-    if command in night_commands:
+        return dict()
+    elif command == 'night':
+        if not user_profile.night_mode:
+            return dict(msg=do_switch_setting(command=command,
+                                              switch_command='day',
+                                              setting='night_mode',
+                                              setting_value=True,
+                                              user_profile=user_profile))
+        return dict(msg='You are still in night mode.')
```

Comment:
I'd swap the order of the cases here; it'd be more readable with the smaller thing returning first.